lib/SILOptimizer/Transforms/RedundantLoadElimination.cpp

```diff
@@ -1523,6 +1662,64 @@ void RLEContext::processBasicBlocksForRLE(bool Optimistic) {
   }
 }
 
+// TODO: Consolidate with CSE and move this to InstOptUtils.cpp
+/// Given a newly created value 'Val', insert destroy_value at its frontiers
+/// when not already found
+static void fixupLifetimeOfValue(ValueLifetimeAnalysis &Vla, SILValue Val) {
+  ValueLifetimeAnalysis::Frontier ValFrontier;
+  Vla.computeFrontier(ValFrontier, ValueLifetimeAnalysis::DontModifyCFG);
```

Comment:
computeFrontier with ValueLifetimeAnalysis::DontModifyCFG may fail in case of critical edges. You might want to use AllowToModifyCFG

Reply:
I thought since we don't have critical edges in OSSA anymore this is not needed. But I added an assert and it fired in a simple control flow with frontier being at the return instruction. I'll look at this further.

Reply:
Ah yes, I didn't realize that we don't have critical edges in OSSA anymore.